spec.html

```diff
@@ -32981,9 +32983,10 @@
         <p>When the `ArrayBuffer` function is called with optional argument _length_, the following steps are taken:</p>
         <emu-alg>
           1. If NewTarget is *undefined*, throw a *TypeError* exception.
-          1. Let _integerLength_ be ToInteger(_length_).
-          1. If _integerLength_ &lt; 0, throw a *RangeError* exception.
-          1. Let _byteLength_ be ? ToLength(_integerLength_).
+          1. Let _numberLength_ be ToNumber(_length_).
+          1. If _numberLength_ &lt; 0, throw a *RangeError* exception.
+          1. Let _byteLength_ be ? ToLength(_numberLength_).
+          1. If SameValueZero(_numberLength_, _byteLength_) is *false*, throw a *RangeError* exception.
```

Comment:
I think with this change, `new ArrayBuffer()` throws a RangeError, because then numberLength is NaN and byteLength is 0.